File: frontend/src/types/emergency.ts
export type FacilityType = "hospital" | "police_station" | "tourist_police";

export interface EmergencyFacility {
  id: string;
  name: string;
  facility_type: FacilityType;
  address: string;
  district: string;
  latitude: number;
  longitude: number;
  phone_number: string | null;
  notes: string | null;
  distance_km: number | null;
}

export interface EmergencyFacilityListResponse {
  items: EmergencyFacility[];
  total: number;
}

export interface EmergencyFacilityCreate {
  name: string;
  facility_type: FacilityType;
  address: string;
  district: string;
  latitude: number;
  longitude: number;
  phone_number?: string | null;
  notes?: string | null;
}

export interface FacilitySearchParams {
  facility_type?: FacilityType;
  district?: string;
  search?: string;
  lat?: number;
  lng?: number;
  limit?: number;
}

export interface EmergencyPhrase {
  id: string;
  english: string;
  bengali: string;
  romanized: string;
}

export interface EmergencyPhraseCategory {
  category: string;
  phrases: EmergencyPhrase[];
}

export interface TranslatePayload {
  text: string;
  dialect?: string;
}

export interface TranslateResponse {
  original_text: string;
  bengali: string;
  romanized: string;
  dialect: string;
  dialect_text: string | null;
  dialect_romanized: string | null;
}

File: backend/app/schemas/emergency.py
import uuid
from datetime import datetime
from typing import List, Optional

from pydantic import BaseModel, ConfigDict, Field


# ── Facility schemas ──


class EmergencyFacilityRead(BaseModel):
    id: uuid.UUID
    name: str
    facility_type: str
    address: str
    district: str
    latitude: float
    longitude: float
    phone_number: Optional[str] = None
    notes: Optional[str] = None
    distance_km: Optional[float] = None  # computed from user location

    model_config = ConfigDict(from_attributes=True)


class EmergencyFacilityListResponse(BaseModel):
    items: List[EmergencyFacilityRead]
    total: int


# ── Phrase schemas ──


class EmergencyPhraseRead(BaseModel):
    id: str
    english: str
    bengali: str
    romanized: str


class EmergencyPhraseCategoryRead(BaseModel):
    category: str
    phrases: List[EmergencyPhraseRead]


# ── Translation schemas ──


class TranslateRequest(BaseModel):
    text: str = Field(..., min_length=1, max_length=1000)
    dialect: Optional[str] = Field(
        None,
        description="Target dialect: sylheti, chittagonian, barisali, rangpuri, or standard (default)",
    )


class TranslateResponse(BaseModel):
    original_text: str
    bengali: str
    romanized: str
    dialect: str
    dialect_text: Optional[str] = None
    dialect_romanized: Optional[str] = None
